### ISSUE TITLE
Ensure solid-server works when installed globally with npm

### ISSUE THREAD

**Issue by RubenVerborgh** [solid/node-solid-server#685]
There are reports of the solid-server executable not working when installed globally. It's rather important that it does, however, or we essentially have a broken npm package.

While this is hard to test on Travis, Solid developers could always run the global executable:
```
npm ls
solid-server start # rather than bin/solid-server
```

**Comment by akuckartz** [comment]
Are there any tests which are run automatically ?

**Comment by RubenVerborgh** [comment]
Yes, but only library tests, no tests that cover the executable.

**Comment by justinwb** [comment]
Running into an issue when doing a clean global install on ubuntu.
npm = 5.6
node = 8.11.3

$ sudo npm install -g solid-server
npm WARN deprecated nodemailer@3.1.8: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
npm ERR! code 1
npm ERR! Command failed: /usr/bin/git clone -q git://github.com/solid/oidc-op.git /home/ubuntu/.npm/_cacache/tmp/git-clone-1fab4c56
npm ERR! /home/ubuntu/.npm/_cacache/tmp/git-clone-1fab4c56/.git: Permission denied
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2018-07-09T17_16_24_769Z-debug.log

**Comment by RubenVerborgh** [comment]
Seems like that specific issue would be solved in develop, as that version no longer depends on a package from GitHub (and it shouldn't).

@kjetilk Shall we release a new patch version?

**Comment by kjetilk** [comment]
Yeah, lets do a patch version! Lets find a time for you to walk me through the release process.

**Comment by RubenVerborgh** [comment]
Okay, ping me tomorrow afternoon!

**Comment by justinwb** [comment]
let me know once it's pushed and i'll confirm the fix 👍

**Comment by RubenVerborgh** [comment]
@justinwb Published, please test :-)

There might be other blockers ahead, but not the git ones.

**Comment by justinwb** [comment]
This is still happening to me, but i believe that i tracked down the root cause, identified in this issue: https://github.com/zkat/pacote/issues/105

I can confirm that when i run the global install as sudo, i get the error error time. However, if I su to root rather than 'sudo -s' or running the command as sudo, the install works correctly.

**Comment by RubenVerborgh** [comment]
> This is still happening to me

The exact same error? Because the dependency on a git repository should be gone.

> zkat/pacote#105

They mention npm 5.x there. Does it also happen if you `sudo npm install -g npm@latest` before?

**Comment by justinwb** [comment]
It was almost exactly the same, although i noticed the flags getting passed to git were slightly different. I should've grabbed them and pasted back in, but I can reproduce exactly later tonight. I was at least able to get around it by su'ing so I'm not blocked, but I'd still consider this a blocking issue since we recommend global install and 99% of people will do that via sudo because they're using cloud instances.  I will test with latest npm tonight.

**Comment by RubenVerborgh** [comment]
Yeah the `su` thing is definitely a blocker; don't know if `nvm` is the easier option, since that might get messy with different users, too.

**Comment by kjetilk** [comment]
Hmmm, this seems like something we should have fixed soon, but are we effectively blocked by the upstream issue?

**Comment by justinwb** [comment]
As some added context for this, here's a log demonstrating the issue exactly with a clean install, as well as testing LTS and latest version of npm. Note that switching from sudo to straight su works at the end...
```
$ sudo -s
# curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
# apt-get install -y nodejs
# apt-get install -y build-essential
# npm -v
5.6.0
# npm install -g solid-server
npm WARN deprecated nodemailer@3.1.8: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
npm ERR! code 1
npm ERR! Command failed: /usr/bin/git clone --depth=1 -q -b rc-0.3.2 git://github.com/solid/oidc-op.git /home/ubuntu/.npm/_cacache/tmp/git-clone-38eb91d4
npm ERR! /home/ubuntu/.npm/_cacache/tmp/git-clone-38eb91d4/.git: Permission denied
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2018-08-12T02_39_01_598Z-debug.log
# npm install -g npm@latest
# npm -v
6.3.0
# npm install -g solid-server
npm WARN deprecated nodemailer@3.1.8: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
npm ERR! code 1
npm ERR! Command failed: /usr/bin/git clone --depth=1 -q -b rc-0.3.2 git://github.com/solid/oidc-op.git /home/ubuntu/.npm/_cacache/tmp/git-clone-86633c9f
npm ERR! /home/ubuntu/.npm/_cacache/tmp/git-clone-86633c9f/.git: Permission denied
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2018-08-12T02_40_50_718Z-debug.log
# exit
$ su
# npm install -g solid-server
SUCCESS
```

**Comment by kjetilk** [comment]
OK, thanks a lot, @justinwb ! 

So, we'll do #749  for 4.1.0. and hope that this is solved somewhere down the road.

**Comment by kjetilk** [comment]
There's a PR upstream now, if anyone wants to try it out.

**Comment by cory2067** [comment]
Not sure if it's some incorrect config on my part, but I haven't had luck with global install:

```
$ sudo npm install -g solid-server
$ solid init
$ solid start
internal/modules/cjs/loader.js:596
    throw err;
    ^

Error: Cannot find module '@trust/keychain'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
    at Function.Module._load (internal/modules/cjs/loader.js:520:25)
    at Module.require (internal/modules/cjs/loader.js:650:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/solid-server/node_modules/@trust/oidc-op/src/Provider.js:8:18)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
```

**Comment by kjetilk** [comment]
Hi @cory2067 ! That's definitly not your fault! We should get #749 ironed out very soon now and release 4.1.2, and then you could try again, as we think it should work.

**Comment by justinwb** [comment]
Can confirm that this problem has been resolved in release v4.1.2-rc.0 (https://github.com/solid/node-solid-server/commit/ae99c390049420ad82f92dde1052635e97c44b50)...

```
ubuntu@inrupt-solid-uat:~$ sudo -s
root@inrupt-solid-uat:~# npm install -g solid-server
npm WARN deprecated nodemailer@3.1.8: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
/usr/bin/solid -> /usr/lib/node_modules/solid-server/bin/solid
+ solid-server@4.1.2-rc.0
added 12 packages, removed 37 packages, updated 18 packages and moved 2 packages in 8.505s
```

@cory2067 - While your issue was not specifically the same as the issue reported here, the fix was, and you should no longer be running into this anymore either with the latest release (v4.1.2-rc.0)
